pkg/generate/generate.go

```diff
@@ -17,7 +17,7 @@ import (
 )
 
 // Generate reads bsf.hcl, resolves dependencies and generates bsf.lock, bsf/flake.nix, bsf/default.nix, etc.
-func Generate(fh *hcl2nix.FileHandlers, sc buildsafev1.SearchServiceClient) error {
+func Generate(fh *hcl2nix.FileHandlers, sc buildsafev1.SearchServiceClient, nameMap map[string]string) error {
```

Comment:
I would suggest to not modify the Generate function's signature for this case. 

We should expect the `conf.OCIArtifact[i].Name` to be modified before it is passed here. cc : @rakshitgondwal